src/frontend/src/lib/api/idb.api.ts

```diff
@@ -1,21 +1,26 @@
 import { browser } from '$app/environment';
+import { ETHEREUM_TOKEN_ID } from '$env/tokens.env';
 import type { IdbEthAddress } from '$lib/types/idb';
+import type { TokenId } from '$lib/types/token';
 import type { Principal } from '@dfinity/principal';
 import { isNullish } from '@dfinity/utils';
 import { createStore, del, get, set, update, type UseStore } from 'idb-keyval';
 
 // There is no IndexedDB in SSG. Since this initialization occurs at the module's root, SvelteKit would encounter an error during the dapp bundling process, specifically a "ReferenceError [Error]: indexedDB is not defined". Therefore, the object for bundling on NodeJS side.
-const oisyEthAddressesStore = browser
-	? createStore('oisy-eth-addresses', 'eth-addresses')
-	: ({} as unknown as UseStore);
+export const idbAddressesStore = (tokenId: TokenId) =>
+	browser
+		? createStore(`oisy-${tokenId.description}-address`, `${tokenId.description}-address`)
```

Comment:
Does it work if the description contains special charaters such as emoji or space?

Reply:
the first one can be a no-restricted string, but the second one is a keypath, so it should not have spaces (or unexpected special characters).

But we don't expect `tokenId` to have spaces and/or emojis, right? Otherwise it's usage as key in type `Record` should be revisited in a few parts of the code

Reply:
We don't but we don't enforce it neither. Let's add it on our todo list or I guess it will be improved the day we migrate to zod.

One more question, why `tokenId` and not `networkId`?

Reply:
the same reason I commented [in the other PR](https://github.com/dfinity/oisy-wallet/pull/2012#discussion_r1715326867): right now, I am thinking that we want to associate it with the crypto behind the chain? but if you think that here it would be stronger/more correct `networkId`, I change it

Reply:
I slept on it, and makes sense that in this case it is the network. Changed a little the code, have a look